sw/host/opentitanlib/src/crypto/spx.rs

```diff
@@ -44,6 +45,38 @@ pub trait SpxPublicKeyPart {
     }
 }
 
+#[derive(Clone)]
+pub enum SpxKey {
+    Public(SpxPublicKey),
+    Private(SpxKeypair),
+}
+
+impl SpxPublicKeyPart for SpxKey {
+    fn pk(&self) -> &spx::PublicKey {
+        match self {
+            SpxKey::Public(k) => k.pk(),
+            SpxKey::Private(k) => k.pk(),
+        }
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum SpxError {
+    #[error("SPHINCS+ key load error\nPublic key: {0}\nKey pair: {1}")]
+    LoadError(anyhow::Error, anyhow::Error),
+}
```

Comment:
I'm not sure if it's better to have this as a distinct error or whether it would be better to use `anyhow::Context` to report on this sort of error condition.

Consider:
```
pub fn load_spx_key(key_file: &Path) -> Result<SpxKey> {
    Ok(match SpxKeypair::read_pem_file(key_file) {
        Ok(sk) => SpxKey::Private(sk),
        Err(e1) => {
            SpxPublicKey::read_pem_file(key_file).context(format!(
                "Could not load {:?} as a public or private key pair\n{:?}", key_file, e1))
        },
    })
}
```

WDYT?

Reply:
Ok.  SGTM.

I confronted a similar problem in key loading for `hsmtool`; I resorted to [inspecting the data](https://github.com/cfrantz/opentitan/blob/hsmtool-util-key/sw/host/hsmtool/src/util/key/rsa.rs#L21-L26) before attempting to decode it.  However, even in that case, I chained the DER-loading together because I didn't want to try to decode enough DER to determine whether the format was PKCS#1 or PKCS#8.